### ISSUE TITLE
GDAL For android

### ISSUE THREAD

**Issue by wanguangtong** [OSGeo/gdal#339]
## Expected behavior and actual behavior.

I use gdal to open image data on android,but it is problem when image data is bigger than 4GB;
and it works normal on windows 

## Steps to reproduce the problem.

I use gdal to open image data on android,but it is problem when image data is bigger than 4GB;
and it works normal on windows 

## Operating system
Android 5.0 ARM

## GDAL version and provenance

GDAL2.2.3

**Comment by mloskot** [comment]
Ensure, you are not hitting the FAT32 file size limit (eg. file on SD card formatted as FAT32). If you do, then head to Android forums and seek for help how to work around it.

**Comment by wanguangtong** [comment]
It is not he FAT32 file size limit,and I can copy 7GB Img data on SD card.Aslo I can read raw data through  C++ File IO,So I think that it is the problem on gdal

**Comment by rouault** [comment]
I've (hopefully: since I just compile tested it) fixed the issue with Android >= 7.0 / API >= 24. Doing it with previous versions look challenging as <stdio.h> functions weren't available for 64-bit operations. Would probably a serious rework of port/cpl_vsil_unix_stdio_64.cpp to use alternate API.

**Comment by varshaj007** [comment]
I used the GDAL Library in my application and it has been working properly. The Google play service announced that they upload only 64-bit compliant APK after 1st Aug 2019. So I rendered the project to 64 bit compliant. Unfortunately, there has some issue happened in GDAL library. they showing that no implementation for the gdal library(ie the .so file according to the ABI type). So I think the corresponding GDAL 64 bit were unavailable for this scope.
So my question is, How we can generate the 64 bit .so files from GDAL Library?
